core/state/statedb.go

```diff
@@ -1297,20 +1351,48 @@ func (s *StateDB) Commit(failPostCommitFunc func(), postCommitFuncs ...func() er
 	var diffLayer *types.DiffLayer
 	var verified chan struct{}
 	var snapUpdated chan struct{}
+	var snapCreated chan common.Hash
+
+	var snapAccountLock sync.Mutex // To protect snapAccount
+
 	if s.snap != nil {
 		diffLayer = &types.DiffLayer{}
 	}
 	if s.pipeCommit {
 		// async commit the MPT
 		verified = make(chan struct{})
 		snapUpdated = make(chan struct{})
+		snapCreated = make(chan common.Hash)
```

Comment:
`snapCreated` can be `chan struct{}`, no need to pass the root back, because we have `s.expectedRoot`